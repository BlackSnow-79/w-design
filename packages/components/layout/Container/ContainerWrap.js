"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
var styled_system_1 = require("styled-system");
var styled_1 = require("../../../utils/styled");
var fixSize_1 = require("../../../utils/fixSize");
exports.ContainerWrap = styled_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  margin-left: auto;\n  margin-right: auto;\n  padding-left: 0.4rem;\n  padding-right: 0.4rem;\n  width: 100%;\n  ", ";\n  max-width: ", ";\n"], ["\n  margin-left: auto;\n  margin-right: auto;\n  padding-left: 0.4rem;\n  padding-right: 0.4rem;\n  width: 100%;\n  ", ";\n  max-width: ", ";\n"])), styled_system_1.color, function (props) { return props.fixSize && fixSize_1.default(props.fixSize); });
exports.ContainerWrap.displayName = "ContainerWrap";
var templateObject_1;
//# sourceMappingURL=ContainerWrap.js.map