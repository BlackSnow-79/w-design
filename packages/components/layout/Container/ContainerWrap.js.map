{"version":3,"file":"ContainerWrap.js","sourceRoot":"","sources":["../../../../src/components/layout/Container/ContainerWrap.tsx"],"names":[],"mappings":";;;;;;AAAA,+CAAsC;AACtC,gDAA2C;AAC3C,kDAA6C;AAOhC,QAAA,aAAa,GAAG,gBAAM,CAAC,GAAG,oNAAA,sHAMnC,EAAK,kBACM,EAAyD,KACvE,KAFG,qBAAK,EACM,UAAC,KAAY,IAAK,OAAA,KAAK,CAAC,OAAO,IAAI,iBAAO,CAAC,KAAK,CAAC,OAAO,CAAC,EAAvC,CAAuC,EACtE;AAEF,qBAAa,CAAC,WAAW,GAAG,eAAe,CAAC"}