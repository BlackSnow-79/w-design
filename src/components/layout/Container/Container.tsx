import * as React from "react";
import styled from "../../../utils/styled";

export interface Props {
  children?: any;
}

const Wrapper = styled.div`
  margin-left: auto;
  margin-right: auto;
  padding-left: 0.4rem;
  padding-right: 0.4rem;
  width: 100%;
`;

const Container = (props: Props) => <Wrapper>{props.children}</Wrapper>;

export default Container;
