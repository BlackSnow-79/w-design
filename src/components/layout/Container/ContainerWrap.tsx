import { color } from "styled-system";
import styled from "../../../utils/styled";
import fixSize from "../../../utils/fixSize";

export interface Props {
  /** xlarge(1280), large(960), medium(840), small(600), xsmall(480) */
  fixSize?: string;
}

export const ContainerWrap = styled.div`
  margin-left: auto;
  margin-right: auto;
  padding-left: 0.4rem;
  padding-right: 0.4rem;
  width: 100%;
  max-width: ${(props: Props) =>
    props.fixSize ? fixSize(props.fixSize) : null};

  ${color};
`;

ContainerWrap.displayName = "ContainerWrap";
