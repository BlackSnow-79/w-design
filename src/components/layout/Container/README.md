Wrapper layout component contains columns layout

```js
<Container style={{ background: "#f3f3f3"}}>
  <div style={{padding: "1em"}}>Full width </div>
</Container>

<br/>

<Container fixSize="xlarge" style={{ background: "#f3f3f3"}}>
  <div style={{padding: "1em"}}>xlarge </div>
</Container>

<br/>

<Container fixSize="large" style={{ background: "#f3f3f3"}}>
  <div style={{padding: "1em"}}>large </div>
</Container>

<br/>

<Container fixSize="medium" style={{ background: "#f3f3f3"}}>
  <div style={{padding: "1em"}}>medium </div>
</Container>

<br/>

<Container fixSize="small" style={{ background: "#f3f3f3"}}>
  <div style={{padding: "1em"}}>small </div>
</Container>

<br/>

<Container fixSize="xsmall" style={{ background: "#f3f3f3"}}>
  <div style={{padding: "1em"}}>xsmall </div>
</Container>
```
