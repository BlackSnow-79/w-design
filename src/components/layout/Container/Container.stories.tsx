import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import Container from ".";

const stories = storiesOf("Container", module);

stories.add(
  "Container",
  withInfo(`The responsive layout.`, { inline: true })(() => (
    <Container>
      <div style={{ background: "#f4f4f4" }}>Child Component</div>
    </Container>
  ))
);
