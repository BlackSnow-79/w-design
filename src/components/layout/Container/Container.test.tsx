import * as React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";
import Container from "./Container";

describe("<Container/>", () => {
  // Render test
  it("should renders correctly", () => {
    const wrapper = shallow(<Container>Children</Container>);
    expect(wrapper).toMatchSnapshot();
  });

  // Props test
  it("should has class", () => {
    const wrapper = shallow(<Container className="test">Children</Container>);
    expect(wrapper.prop("className")).toEqual("test");
  });
});
