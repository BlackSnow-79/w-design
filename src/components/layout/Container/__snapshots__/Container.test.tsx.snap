// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container/> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(ContainerEl)>
    Children
  </Styled(ContainerEl)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Children",
      "className": "sc-bdVaJa iiNbcj",
    },
    "ref": null,
    "rendered": "Children",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Children",
        "className": "sc-bdVaJa iiNbcj",
      },
      "ref": null,
      "rendered": "Children",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
